lldb/source/Plugins/SymbolLocator/Debuginfod/SymbolLocatorDebuginfod.cpp

```diff
@@ -114,29 +177,85 @@ SymbolLocator *SymbolLocatorDebuginfod::CreateInstance() {
 
 static std::optional<FileSpec> GetFileForModule(
     const ModuleSpec &module_spec,
-    std::function<llvm::Expected<std::string>(llvm::object::BuildIDRef)>
-        PullFromServer) {
+    std::function<llvm::StringRef(llvm::object::BuildID)> UrlBuilder,
+    bool force_lookup) {
+  PluginProperties &plugin_props = GetGlobalPluginProperties();
+  llvm::Expected<llvm::StringRef> CacheDirectoryPathOrErr =
+      plugin_props.GetCachePath();
+  // We really do require a cache location to store stuff.
+  if (!CacheDirectoryPathOrErr)
+    return {};
+  llvm::StringRef CacheDirectoryPath = *CacheDirectoryPathOrErr;
+  llvm::SmallVector<llvm::StringRef> DebuginfodUrls =
+      llvm::getDefaultDebuginfodUrls();
```

Comment:
Shouldn't we get this from the setting instead of from `llvm::getDefaultDebuginfodUrls()`?

Reply:
The setting pushes the values into this location (see ServerURLsChangedCallback)

Reply:
The name of that API isn't great. I should remove "Default" from the name.